provider/shim/shim.go

```diff
@@ -0,0 +1,10 @@
+package shim
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	googleWorkspace "github.com/hashicorp/terraform-provider-googleworkspace/internal/provider"
+)
+
+func NewProvider(version string) *schema.Provider {
+	return googleWorkspace.New(version)()
```

Comment:
@stack72 This is what I told you in Slack, the `New` method in the provider [returns a function](https://github.com/hashicorp/terraform-provider-googleworkspace/blob/main/internal/provider/provider.go#L49), differently from the `azuread` example I was taking as reference.

I'm wondering if the source of the `panic: cannot create context from nil parent` is these lines: https://github.com/hashicorp/terraform-provider-googleworkspace/blob/main/internal/provider/provider.go#L198-L199.